backend.py

```diff
@@ -403,8 +403,10 @@ def add_repos(main_repositories, update_repositories, repos):
         for i in answers_pristine['sources']:
             repos = repository.repositoriesFromDefinition(i['media'], i['address'])
             add_repos(main_repositories, update_repositories, repos)
-            repo_gpgcheck = answers.get('repo-gpgcheck', True)
-            gpgcheck = answers.get('gpgcheck', True)
+            repo_gpgcheck = (answers.get('repo-gpgcheck', True) if i['repo_gpgcheck'] is None
+                             else i['repo_gpgcheck'])
+            gpgcheck = (answers.get('gpgcheck', True) if i['gpgcheck'] is None
+                        else i['gpgcheck'])
             for repo in repos:
                 if repo in main_repositories:
```

Comment:
Are all sources mentioned in the answerfile as a `<source>` in `main_repositories`?

Reply:
Assumption was indeed that `main_repositories` comes from `<source>`'s and `update_repositories` from `<driver-source>`'s.  But that's wrong, as supplemental packs can be specified as `<source>` (as documented in `answerfile.txt`).

In more details:

## The workflow goes:
1. parsing
  * `Answerfile.parseSource()`: `<source>` -> `answers['sources']` (named `results` during parsing, `answers` starting from `backend.performInstallation()`)
  * `Answerfile.parseDriverSource()`: `<driver-source>` -> `answers['extra-repos']`
1. (bis) TUI
  * "load driver" screen adds to `answers['extra-repos']` (and does install packages inside initramfs in a non-obvious process, but that's another story, see `driver.doInteractiveLoadDriver()`)
  * the "supplemental packs installation" step at the end of install process XXXX
3. `Repository` objects creation
  * everyone goes through `repository.repositoriesFromDefinition()` and `add_repos()` regardless of "sources" vs. "extra-repos"
4. `add_repos()` separates `UpdateYumRepository`'s (i.e. those with `update.xml`) from the rest, into resp.  `update_repositories`  and `main_repositories`

Essential info: supplemental packs (which include driver packs) all include an `update.xml` (as generated by the DDK's `build-update` script).

So at this point:
- we know the install media `<source>` has no `update.xml`, so it is classified as `main_repository`
- all supplemental packs, whether from `<driver-source>` or from `<source>`, are classified as `update_repository

## What this means for this patch series:
- no supplemental pack is gpgcheck'd as per upstream code (even though some do include a sig! (e.g. `driver-microsemi-aacraid`)
- `<driver-source>` is supposed to point to supplemental packs only, so not having `*gpgcheck` attributes is not a problem until we want to allow checking sigs for them
- a `<source>` element can point declare a source with both "main repos" and supplemental packs; the latter will just ignore the `*gpgcheck` attributes.  This can probably be the case if we build an install media containing supplemental packs alongside the main repo.

## Bottom line:

* we could make the doc a little more clear about supplemental packs not honoring those attributes
* we probably want to add gpgcheck support to supplemental packs too!

Reply:
Thanks for the detailed explanation.

Reply:
Updated the doc to document the behaviour wrt. Supplemental Packs.  Maybe we'll want to find a wording to make it sound less scary that Supplemental Packs are not gpgcheck'd ?